pkg/sentry/platform/ring0/defs.go

```diff
@@ -95,6 +95,9 @@ type SwitchOpts struct {
 	// saved and restored.
 	FloatingPointState *byte
 
+	// ArchReg is the arch-specific register state: not used on amd64.
+	ArchReg *uint64
```

Comment:
I still don't like this ArchReg. Let me try one idea.

Reply:
I think we can move arch specific registers in the Registers struct:

type Registers struct {
       ptRegs linux.PtraceRegs
       TPIDR_EL0 uint64
}

func (r *Registers) PtraceRegs() *linux.PtraceRegs {
       return &r.ptRegs
}

Here is a RFC version for x86: https://github.com/avagin/gvisor/commit/8a6d34cd46681f1a98480c59432d4475b8622271

Reply:
Hi @avagin 
Done.
I also removed TPValue from context64 in this PR.
By the way. I have to put your changes in this PR, otherwise I can't compile it on x86.